src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -216,6 +216,52 @@ public void CheckForBrokerAccountInfoInAccountManager(IDictionary<string, string
             throw new MsalUiRequiredException(MsalError.NoAndroidBrokerAccountFound, MsalErrorMessage.NoAndroidBrokerAccountFound);
         }
 
+        /// <summary>
+        /// This method will acquire all of the accounts in the account manager that have an access token for the given client ID.
+        /// </summary>
```

Comment:
why is this by client id?

Reply:
because the broker separates application caches by client ID. so you need to pass in the client id to get all of the accounts for the current application.